tests/mu_different_test.c

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include <string.h>
+#include "test_helper.h"
+
+#include "macro_utils/macro_utils.h"
+
+#include "mu_different_test.h"
+
+int run_mu_different_tests(void)
+{
+    int result = 0;
+
+    POOR_MANS_ASSERT(MU_DIFFERENT(-2) != -22);
+
+    POOR_MANS_ASSERT(MU_DIFFERENT(-1) != -1);
+
+    POOR_MANS_ASSERT(MU_DIFFERENT(0) != 0);
+
+    POOR_MANS_ASSERT(MU_DIFFERENT(1) != 1);
+
+    POOR_MANS_ASSERT(MU_DIFFERENT(2) != 2);
+
+    POOR_MANS_ASSERT(
+        (MU_DIFFERENT(2, 3) != 2) &&
+        (MU_DIFFERENT(2, 3) != 3)
+    );
+
+    POOR_MANS_ASSERT(
+        (MU_DIFFERENT(3, 2) != 2) &&
+        (MU_DIFFERENT(3, 2) != 3)
+    );
+
+    POOR_MANS_ASSERT(
+        (MU_DIFFERENT(0, 1) != 0) &&
+        (MU_DIFFERENT(0, 1) != 1)
+    );
+
+    POOR_MANS_ASSERT(
+        (MU_DIFFERENT(1, 0) != 0) &&
+        (MU_DIFFERENT(1, 0) != 1)
+    );
+
+    POOR_MANS_ASSERT(
+        (MU_DIFFERENT(0, 1, 2) != 0) &&
+        (MU_DIFFERENT(0, 1, 2) != 1) &&
+        (MU_DIFFERENT(0, 1, 2) != 2)
+    );
+
+    /*works with floats too*/
+    POOR_MANS_ASSERT(
+        (MU_DIFFERENT(0.5) != 0.5)
+    );
+
+    POOR_MANS_ASSERT(
+        (MU_DIFFERENT(0.5, 0) != 0.5) &&
+        (MU_DIFFERENT(0.5, 0) != 0)
+    );
```

Comment:
Consider writing a test with some functions being called. Yes, I know we said side effects are not advised, but for the fun of it.